frontend/src/components/DragAndDrop/DragAndDrop.js

```diff
@@ -0,0 +1,144 @@
+import { useState, useRef, useEffect } from "react";
+import { CloudArrowUpFill } from "react-bootstrap-icons";
+import { useSpring, animated } from "react-spring";
+import "./DragAndDrop.css";
+import NavBar from "../NavBar";
+import App from "../../App";
+
+function DragAndDrop() {
+  const [dragActive, setDragActive] = useState(false);
+  const [fileUploaded, setFileUploaded] = useState(false);
+  const [invalidJson, setInvalidJson] = useState(false);
+  const [lines, setLines] = useState();
+  const inputRef = useRef(null);
+
+  const flyingAnimation = useSpring({
+    transform: fileUploaded ? "translateY(-100%)" : "translateY(0%)",
+  });
+
+  const handleDrag = function (e) {
+    e.preventDefault();
+    e.stopPropagation();
+    if (e.type === "dragenter" || e.type === "dragover") {
+      setDragActive(true);
+    } else if (e.type === "dragleave") {
+      setDragActive(false);
+    }
+  };
+
+  const handleDrop = function (e) {
+    e.preventDefault();
+    e.stopPropagation();
+    setDragActive(false);
+    if (e.dataTransfer.files && e.dataTransfer.files[0]) {
+      const file = e.dataTransfer.files[0];
+      handleFiles(file);
+      setFileUploaded(true);
+    }
+  };
+
+  const handleChange = function (e) {
+    e.preventDefault();
+    e.stopPropagation();
+    if (e.target.files && e.target.files[0]) {
+      const file = e.target.files[0];
+      handleFiles(file);
+      setFileUploaded(true);
+    }
+  };
+
+  const onButtonClick = () => {
+    inputRef.current.value = "";
+    inputRef.current.click();
+  };
+
+  const handleFiles = (file) => {
+    const fileName = file.name;
+    if (fileName.endsWith(".json") || fileName.endsWith(".jsonl")) {
+      const reader = new FileReader();
+      reader.onload = (e) => {
+        try {
+          const dataObjectsArray = e.target.result.trim().split(/\n(?=\{)/);
```

Comment:
Is there a reason for searching for a newline and `{` on next position? Can we just change this regex to `\n`?

Reply:
Yes, you're right. If the data is valid line-based json, then we can skip that extra search.